cpp/src/Slice/Grammar.y

```diff
@@ -1674,41 +1689,33 @@ enumerator
 : ICE_IDENTIFIER
 {
     auto ident = dynamic_pointer_cast<StringTok>($1);
-    auto ens = make_shared<EnumeratorListTok>();
-    ContainerPtr cont = currentUnit->currentContainer();
-    EnumeratorPtr en = cont->createEnumerator(ident->v, nullopt);
-    if (en)
-    {
-        ens->v.push_front(en);
-    }
-    $$ = ens;
+    EnumPtr cont = dynamic_pointer_cast<Enum>(currentUnit->currentContainer());
+    $$ = cont->createEnumerator(ident->v, nullopt);
 }
 | ICE_IDENTIFIER '=' enumerator_initializer
 {
     auto ident = dynamic_pointer_cast<StringTok>($1);
-    auto ens = make_shared<EnumeratorListTok>();
-    ContainerPtr cont = currentUnit->currentContainer();
+    EnumPtr cont = dynamic_pointer_cast<Enum>(currentUnit->currentContainer());
     auto intVal = dynamic_pointer_cast<IntegerTok>($3);
     if (intVal)
     {
         if (intVal->v < 0 || intVal->v > std::numeric_limits<int32_t>::max())
         {
             currentUnit->error("value for enumerator `" + ident->v + "' is out of range");
```

Comment:
What happens after you issue this error?

My understanding is the existing code tried to keep going.

Reply:
Yes, the new code also keeps going. On the line right under this we set:
`$$ = cont->createEnumerator(ident->v, static_cast<int>(intVal->v));`
this is effectively a return statement in Bison. So we still construct the enumerator, and return it.